internal/scorer/algorithm/value_test.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package algorithm
+
+import (
+	"testing"
+)
+
+func TestExistsCondition(t *testing.T) {
+	tests := []struct { //nolint:govet
+		name   string
+		f      Field
+		fields map[string]float64
+		want   bool
+	}{
+		{
+			name:   "exists",
+			f:      Field("a"),
+			fields: map[string]float64{"a": 1},
+			want:   true,
+		},
+		{
+			name:   "not exists",
+			f:      Field("a"),
+			fields: map[string]float64{"b": 1},
+			want:   false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got := ExistsCondition(test.f)
+
+			if got(test.fields) != test.want {
+				t.Errorf("ExistsCondition() = %v, wantVal %v", got(test.fields), test.want)
+			}
+		})
+	}
+}
+
+func TestNotCondition(t *testing.T) {
+	tests := []struct { //nolint:govet
+		name   string
+		f      Field
+		fields map[string]float64
+		want   bool
+	}{
+		{
+			name:   "exists but wantVal false",
+			f:      Field("a"),
+			fields: map[string]float64{"a": 1},
+			want:   false,
+		},
+		{
+			name:   "not exists but wantVal true",
+			f:      Field("a"),
+			fields: map[string]float64{"b": 1},
+			want:   true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			exists := ExistsCondition(test.f)
+			got := NotCondition(exists)
+
+			if got(test.fields) != test.want {
+				t.Errorf("NotCondition() = %v, wantVal %v", got(test.fields), test.want)
+			}
+		})
+	}
+}
+
+func TestField_Value(t *testing.T) {
```

Comment:
I have removed `TestField_Value` but kept and renamed `TestConditionalValue_Value` because the conditional value calls the field value inside of it. I have renamed `TestConditionalValue_Value` to `TestValue`. I have also changed `"f"` to `"value"` and made a structure for want. Thank You!